### PR TITLE
Feat/create triangle primitive

### PR DESCRIPTION
**Pull request by mathieurobert1**
I everyone ! I just finish to create the triangle primitive. Hope you enjoy.
Screenshot from 2024-05-11 10-16-48
image

### CODE REVIEW

File: include/Primitives/Triangle.hpp [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+             *
+             */
+            void createNormals();
+
+            /**
+             * @brief Get the Triangle Normal object.
+             *
+             * @return Math::Vector3D
+             */
+            Math::Vector3D getTriangleNormal();
```
Comment (AxelF44) [inline include/Primitives/Triangle.hpp:134]:
```suggestion
            Math::Vector3D getTriangleNormal() const;
```

File: include/Primitives/Triangle.hpp [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+            /**
+             * @brief Get the Triangle Inverse Normal object.
+             *
+             * @return Math::Vector3D
+             */
+            Math::Vector3D getTriangleInverseNormal();
```
Comment (AxelF44) [inline include/Primitives/Triangle.hpp:141]:
```suggestion
            Math::Vector3D getTriangleInverseNormal() const;
```

File: include/Primitives/Triangle.hpp [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+            /**
+             * @brief Set the Triangle Normal object.
+             *
+             * @param normal to set
+             */
+            void setTriangleNormal(const Math::Vector3D& normal);
+
+            /**
+             * @brief Set the Triangle Inverse Normal object
```
Comment (AxelF44) [inline include/Primitives/Triangle.hpp:151]:
```suggestion
             * @brief Set the Triangle Inverse Normal object.
```

File: src/Plugins/Primitives/Triangle/Triangle.cpp [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+{
+    return _vertex2;
+}
+
+Math::Point3D Primitive::Triangle::getVertex3() const
+{
+    return _vertex3;
+}
+
+Math::Vector3D Primitive::Triangle::getTriangleNormal()
```
Comment (AxelF44) [inline src/Plugins/Primitives/Triangle/Triangle.cpp:140]:
```suggestion
Math::Vector3D Primitive::Triangle::getTriangleNormal() const
```

File: src/Plugins/Primitives/Triangle/Triangle.cpp [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+{
+    return _normal;
+}
+
+Math::Vector3D Primitive::Triangle::getTriangleInverseNormal()
```
Comment (AxelF44) [inline src/Plugins/Primitives/Triangle/Triangle.cpp:145]:
```suggestion
Math::Vector3D Primitive::Triangle::getTriangleInverseNormal() const
```

File: src/Plugins/Primitives/Triangle/Triangle.cpp [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+
+    collider.maxX.first = false;
+    collider.minX.first = false;
+    collider.maxY.first = false;
+    collider.minY.first = false;
+    collider.maxZ.first = false;
+    collider.minZ.first = false;
+
+    return collider;
+}
```
Comment (AxelF44) [inline src/Plugins/Primitives/Triangle/Triangle.cpp:179]:
```suggestion
}

```

**Review by AxelF44 — CHANGES_REQUESTED** [review]

**Review by AxelF44 — DISMISSED** [review]

File: include/Primitives/Triangle.hpp [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+
+            /**
+             * @brief Get the Triangle Normal object.
+             *
+             * @return Math::Vector3D
+             */
+            Math::Vector3D getTriangleNormal() const;
+
+            /**
+             * @brief Get the Triangle Inverse Normal object.
```
Comment (Njord201) [inline include/Primitives/Triangle.hpp:137]:
```suggestion
             * @brief Get the triangle inverse normal value.
```

File: include/Primitives/Triangle.hpp [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+             *
+             * @return Math::Vector3D
+             */
+            Math::Vector3D getTriangleInverseNormal() const;
+
+            /**
+             * @brief Set the Triangle Normal object.
+             *
+             * @param normal to set
```
Comment (Njord201) [inline include/Primitives/Triangle.hpp:146]:
```suggestion
             * @brief Set the triangle normal value.
             *
             * @param normal new value to set
```

File: include/Primitives/Triangle.hpp [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+             */
+            void setTriangleNormal(const Math::Vector3D& normal);
+
+            /**
+             * @brief Set the Triangle Inverse Normal object.
+             *
+             * @param inverseNormal to set
```
Comment (Njord201) [inline include/Primitives/Triangle.hpp:153]:
```suggestion
             * @brief Set the triangle inverse normal value.
             *
             * @param inverseNormal new value to set
```

**Review by Njord201 — CHANGES_REQUESTED** [review]
Some request on the documentation

**Review by Njord201 — APPROVED** [review]

**Review by AxelF44 — APPROVED** [review]

**Pull request merged**